src/commands/functions/create.js

```diff
@@ -196,23 +195,76 @@ const pickTemplate = async function () {
 
 const DEFAULT_PRIORITY = 999
 
-/* get functions dir (and make it if necessary) */
-const ensureFunctionDirExists = function (context, flags, config) {
-  const functionsDir = config.build && config.build.functions
-  if (!functionsDir) {
-    context.log(`${NETLIFYDEVLOG} No functions folder specified in netlify.toml`)
-    process.exit(1)
+/**
+ * Get functions dir (and make it if necessary)
+ * @this FunctionsCreateCommand
+ * @returns {string | never} - functions directory or throws an error
+ */
+const ensureFunctionDirExists = async function () {
+  const { api, config, site } = this.netlify
+  const { log } = this
+  let functionsDirHolder = config.build && config.build.functions
+  let siteData = null
+  const siteId = site.id
+
+  // try to find functions_dir in site settings if not in user's netlify config
+  if (!functionsDirHolder) {
```

Comment:
I don't think this part is required as we already retrieve site information when resolving the configuration in the `Command` base class.

Reply:
Removed the one where we retrieve site info thanks!